### PR TITLE
Suma product tree

### PR DESCRIPTION
**Pull request by tmuntaner**

### CODE REVIEW

File: lib/rmt/mirror.rb [hunk 1]
```diff
@@ -20,6 +20,26 @@ def initialize(mirroring_base_dir: RMT::DEFAULT_MIRROR_DIR, logger:, mirror_src:
...
+    base_dir = File.expand_path(File.join(@mirroring_base_dir, '/../')) if @mirroring_base_dir == RMT::DEFAULT_MIRROR_DIR
+
+    @repository_dir = File.join(base_dir, '/suma/')
+    @downloader.repository_url = URI.join(repository_url)
+    @downloader.destination_dir = @repository_dir
+
+    @logger.info _('Mirroring SUMA product tree to %{dir}') % { dir: @repository_dir }
+    @downloader.download('product_tree.json')
+  rescue RMT::Downloader::Exception => e
+    raise RMT::Mirror::Exception.new(_('Could not mirror suma product tree with error: %{error}') % { error: e.message })
```
Comment (felixsch) [inline lib/rmt/mirror.rb]: Can we not just log the error here and remove all try/catch statements in the other files?

File: package/nginx-http.conf [hunk 2]
```diff
@@ -20,5 +20,9 @@ server {
         return 301 https://$host$request_uri;
     }
 
+    location /suma {
```
Comment (felixsch) [inline package/nginx-http.conf]: Is the data required in plain `http`? I thought SM is also consuming other endpoints which are all only available via `https`?

File: public/suma/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+*
```
Comment (felixsch) [inline public/suma/.gitignore]: I like this! Instead of have a `.keep` file around add a `.gitignore` :+1:

File: package/nginx-http.conf (hunk 2 above)
Comment (tmuntaner) [reply package/nginx-http.conf]: It is a request that RMT works more with http and that https is just the better workflow.

See: https://trello.com/c/fGuUG7O1/3105-rmt-allow-registration-over-http

Comment (felixsch) [reply package/nginx-http.conf]: Okay didn't know about this problem. Than I'm fine with this (not with with autoyast being unable to import certs...)

File: lib/rmt/mirror.rb (hunk 1 above)
Comment (tmuntaner) [reply lib/rmt/mirror.rb]: I guess I feel not having the function a black box and that we let the cli itself handle the failure state.

Also, it should make solving issues like [this](https://github.com/SUSE/rmt/issues/334) easier. I think that we actually want to implement a summary of failures these cli commands.

**Comment by felixsch** [comment]
While building the package I noticed that a version bump is missing

**Comment by tmuntaner** [comment]
Shouldn't we only bump the version when we do a submit request?

**Pull request merged**